clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -89,23 +89,31 @@ void asuint(double4, out uint4, out uint4);
 /// \param X The X input value.
 /// \param Y The Y input value.
 
+template <typename T>
 _HLSL_16BIT_AVAILABILITY(shadermodel, 6.2)
-const inline half distance(half X, half Y) {
+const inline __detail::enable_if_t<__detail::is_arithmetic<T>::Value &&
+                                       __detail::is_same<half, T>::value,
```

Comment:
capitalize value here? I assume this didn't cause you an error but nice to be consistent with use of Caps

Reply:
yeah this was weird but `is_same` was lowercase value. didn't change it because thought it was out of scope of this change, but might do a cleanup pr after.

Reply:
Why is it out of the scope if this change if you added it? Sorry I don't understand.

Reply:
I didn't add `is_same` thats been in `clang/lib/Headers/hlsl/hlsl_detail.h` for a long time. predating most the other stuff like `enable_if_t` and `is_arithmetic`.

Reply:
Agree it makes sense to leave as a cleanup, but we should use lower case `value` here rather than llvm conventions since these helpers match the standard C++ versions of the same.